Dalamud/Interface/Internal/UiDebug2/Browsing/TimelineTree.cs

```diff
@@ -0,0 +1,386 @@
+
+        var textColorColumn = new KeyGroupColumn<ByteColor>("Text Color", PrintRGB);
+
+        for (var f = 0; f < keyGroup.KeyFrameCount; f++)
+        {
+            textColorColumn.Add(keyGroup.KeyFrames[f].Value.RGB);
+        }
+
+        columns.Add(textColorColumn);
+    }
+
+    private static void GetPartIdColumn(AtkTimelineKeyGroup keyGroup, List<IKeyGroupColumn> columns)
+    {
+        if (keyGroup.KeyFrameCount <= 0)
+        {
+            return;
+        }
+
+        var partColumn = new KeyGroupColumn<ushort>("Part ID");
+
+        for (var f = 0; f < keyGroup.KeyFrameCount; f++)
+        {
+            partColumn.Add(keyGroup.KeyFrames[f].Value.UShort);
+        }
+
+        columns.Add(partColumn);
+    }
+
+    private static void GetEdgeColumn(AtkTimelineKeyGroup keyGroup, List<IKeyGroupColumn> columns)
+    {
+        if (keyGroup.KeyFrameCount <= 0)
+        {
+            return;
+        }
+
+        var edgeColorColumn = new KeyGroupColumn<ByteColor>("Edge Color", PrintRGB);
+
+        for (var f = 0; f < keyGroup.KeyFrameCount; f++)
+        {
+            edgeColorColumn.Add(keyGroup.KeyFrames[f].Value.RGB);
+        }
+
+        columns.Add(edgeColorColumn);
+    }
+
+    private static void GetLabelColumn(AtkTimelineKeyGroup keyGroup, List<IKeyGroupColumn> columns)
+    {
+        if (keyGroup.KeyFrameCount <= 0)
+        {
+            return;
+        }
+
+        var labelColumn = new KeyGroupColumn<ushort>("Label");
+
+        for (var f = 0; f < keyGroup.KeyFrameCount; f++)
+        {
+            labelColumn.Add(keyGroup.KeyFrames[f].Value.Label.LabelId);
+        }
+
+        columns.Add(labelColumn);
+    }
+
+    private static void PrintRGB(ByteColor c) => PrintColor(c, $"0x{SwapEndianness(c.RGBA):X8}");
+
+    private static void PrintAlpha(byte b) => PrintColor(new Vector4(b / 255f), PadEvenly($"{b}", 25));
+
+    private static void PrintAddCell(Vector3 add)
+    {
+        var fmt = PadEvenly($"{PadEvenly($"{add.X}", 30)}{PadEvenly($"{add.Y}", 30)}{PadEvenly($"{add.Z}", 30)}", 100);
+        PrintColor(new Vector4((add / new Vector3(510f)) + new Vector3(0.5f), 1), fmt);
+    }
+
+    private static void PrintMultiplyCell(ByteColor byteColor)
+    {
+        var multiply = new Vector3(byteColor.R, byteColor.G, byteColor.B);
+        var fmt = PadEvenly($"{PadEvenly($"{multiply.X}", 25)}{PadEvenly($"{multiply.Y}", 25)}{PadEvenly($"{multiply.Z}", 25)}", 100);
+        PrintColor(multiply / 255f, fmt);
+    }
+
+    private static string PadEvenly(string str, float size)
+    {
+        while (ImGui.CalcTextSize(str).X < size * ImGuiHelpers.GlobalScale)
+        {
+            str = $" {str} ";
+        }
+
+        return str;
+    }
+
+    private void PrintAnimation(AtkTimelineAnimation animation, int a, bool isActive, nint address)
+    {
+        var columns = this.BuildColumns(animation);
+
+        var col = ImRaii.PushColor(ImGuiCol.Text, isActive ? new Vector4(1, 0.65F, 0.4F, 1) : new(1));
```

Comment:
You're doing this all over so this is more of a general comment: ImRaii doesn't really do anything for safety if you're not also using it with a using block or statement. Shouldn't be very difficult to fix!

Reply:
It's also possible to pass the `isActive` to the condition parameter, so it only pushes the color when isActive is true:
```cs
using var col = ImRaii.PushColor(ImGuiCol.Text, new Vector4(1, 0.65F, 0.4F, 1), isActive);
```
In case isActive is false, it will not push the color and continue to use the current active one.
Just wanted to note this, but I didn't check context if it's necessary here to push a white color or not. 🙂